controlplane/pkg/nsmd/network_service_server.go

```diff
@@ -67,7 +67,8 @@ func (srv *networkServiceServer) Request(ctx context.Context, request *networkse
 	dpCtx, dpCancel := context.WithTimeout(context.Background(), nseConnectionTimeout)
 	defer dpCancel()
 
-	endpoint, err := srv.model.SelectEndpoint(request.GetConnection().GetNetworkService())
+	labels := request.GetConnection().GetLabels()
+	endpoint, err := srv.model.SelectEndpoint(request.GetConnection().GetNetworkService(), &labels)
```

Comment:
@edwarnicke `request.GetConnection().GetNetworkService()` returns the `string` name of the requested service.
In order to get `registry.NetworkService` as you suggest, I need to execute some lookup. I guess this could be the model's `GetNetworkServiceEndpoints`, but that returns a slice.
Or am I getting it completely wrong?